### ISSUE TITLE
Stuck with multi-line control structure

### ISSUE THREAD

**Issue by pierrocknroll** [squizlabs/PHP_CodeSniffer#2670]
Hello,

When using the PSR2 and PSR-12 together (phpcs version 3.5.1), it's impossible to fix this code  :
```php
if (
    $cond1
    && $cond2
) {
    ...
}
```
It gives me this error : `Expected 0 spaces after opening bracket; newline found`

When I alter the code as follows then the error is : `The first expression of a multi-line control structure must be on the line after the opening parenthesis`

```php
if ($cond1
    && $cond2
) {
    ...
}
```

I'm totally stuck :D
How can I do to fix this ?
Thanks

**Comment by jrfnl** [comment]
Duplicate of #2113 ?

**Comment by gsherwood** [comment]
PSR2 and PSR12 have slightly different rules for how conditions should be written.

Both PSR2 and PSR12 state that:
> There MUST NOT be a space after the opening parenthesis

But PSR12 specifically allows for expressions to be split over multiple lines:
> Expressions in parentheses MAY be split across multiple lines, where each subsequent line is indented at least once. When doing so, the first condition MUST be on the next line.

Following PSR12 here would violate PSR2.

PSR12 is an updated version of PSR2 and incorporates almost all of PSR2 and PSR1 in it. Just use PSR12 - not both PSR12 and PSR2 together.

**Comment by grantlucas** [comment]
> not both PSR12 and PSR2 together

Ahhhh, for my setup at least, _that_ was my issue. I _added_ PSR-12 but didn't think to _remove_ PSR-2.

**Comment by pierrocknroll** [comment]
Well, it's not clearly said in the PSR-12 that it includes 1 & 2.
I don't see why they would do that.

**Comment by gsherwood** [comment]
> Well, it's not clearly said in the PSR-12 that it includes 1 & 2.

The first line of the overview reads:
> This specification extends, expands and replaces PSR-2, the coding style guide and requires adherence to PSR-1, the basic coding standard.

The rest of the overview is entirely about PSR-2: https://www.php-fig.org/psr/psr-12/#overview

There is a lot of ambiguity in PSR-12, but I don't think it's relationship to PSR-2 is left unclear.

**Comment by fmagrosoto** [comment]
You can exec the linter test as:
./vendor/bin/phpcs --standard=PSR2 --colors ./your/directory
